### ISSUE TITLE
[Feature Request] Android: Option to pause playback on audio becoming noisy

### ISSUE THREAD

**Issue by mattleff** [doublesymmetry/react-native-track-player#1545]
**What is the need and use case of this feature?**
In Android pausing audio when a headset or bluetooth disconnects is a standard playback behavior. This is accomplished by listening to the [`ACTION_AUDIO_BECOMING_NOISY`](https://developer.android.com/guide/topics/media-apps/volume-and-earphones#becoming-noisy) intent and pausing audio.

**Describe the ideal solution**
It would be great if there was a RNTP option something like `pausePlaybackOnAudioBecomingNoisy: boolean`.

**Describe alternatives you've considered**
I'm not sure there's any other workarounds possible.

**How I can Help**
I'd be happy to work on a PR although I believe it also requires a change in the KotlinAudio repo.

**Comment by jspizziri** [comment]
@mattleff this seems like something that we'd want to support. My question would be if this should be standard behavior or if it should be configurable. We'd also want feature parity on iOS, have you looked into that side at all?

@dcvz thoughts? Basically auto pause when disconnecting an audio device like headphones.

**Comment by mattleff** [comment]
@jspizziri Good point about iOS. It appears that currently iOS pauses playback automatically whenever an output channel is disconnected. So that would be an argument to have it be standard for Android too.

**Comment by jspizziri** [comment]
@mattleff do you need any direction before opening a PR? Probably just for the unconfigurable option.

**Comment by mattleff** [comment]
@jspizziri I've opened a simple PR for this: https://github.com/doublesymmetry/KotlinAudio/pull/26 I'm not sure to how to test it, so any guidance there would be appreciated.

**Comment by jspizziri** [comment]
Nice, at this point we'll need to wait for feedback from the DoubleSymmetry team. TBH, testing updates to KotlinAudio within the context of this project is something I've not figured out yet. However, I've requested some documentation from those in the know which, once I have, will be added to this repository.

I'm quite the n00b when it comes to Gradle, but at a high level what I've been told is that you can publish your local modifications to a local maven repository and then install the local version in this package. If that means anything to you please run with it. If I don't get docs in a few weeks I'll probably try to figure it out myself again (but that didn't go well last time ha).
